### PR TITLE
Fixes Issue #515 - Properly detect MacOSX

### LINKED ISSUE #869: Tests that use directory separator char to determine platform misreport Linux on MaxOSX
After fixing #515 so that MacOSX is correctly returned when running on a Mac, the three tests fail because they are using the directory separator char to determine the platform and setting it to Linux where it could be MacOSX.

To get the tests passing on Mac, I added a platform attribute for now. These should be fixed properly when we have more time.

```
1) Failed : NUnit.Framework.Attributes.ApplyToTestTests.PlatformAttributeRunsTest
  Expected: Runnable
  But was:  Skipped
at NUnit.Framework.Attributes.ApplyToTestTests.PlatformAttributeRunsTest () [0x00044] in /Users/rob/src/nunit/src/NUnitFramework/tests/Attributes/ApplyToTestTests.cs:363

2) Failed : NUnit.Framework.Attributes.TestCaseAttributeTests.CanExcludePlatform
  Expected: Skipped
  But was:  Runnable
at NUnit.Framework.Attributes.TestCaseAttributeTests.CanExcludePlatform () [0x00068] in /Users/rob/src/nunit/src/NUnitFramework/tests/Attributes/TestCaseAttributeTests.cs:313

3) Failed : NUnit.Framework.Attributes.TestCaseAttributeTests.CanIncludePlatform
  Expected: Runnable
  But was:  Skipped
at NUnit.Framework.Attributes.TestCaseAttributeTests.CanIncludePlatform () [0x00068] in /Users/rob/src/nunit/src/NUnitFramework/tests/Attributes/TestCaseAttributeTests.cs:291
```

### PR DESCRIPTION
**Pull request by rprouse**
Also fixes #869

### CODE REVIEW

File: src/NUnitFramework/tests/Internal/PlatformDetectionTests.cs [hunk 1]
```diff
@@ -281,6 +281,15 @@ public void DetectMacOSX()
                 new OSPlatform(PlatformID.MacOSX, new Version(0, 0)),
                 "MacOSX");
         }
+
+#if !PORTABLE
+        [Test]
+        [Platform(Include = "MacOSX")]
+        public void CurrentPlatformIsMacOSX()
+        {
+            Assert.That(OSPlatform.CurrentPlatform.IsMacOSX, Is.True);
```
Comment (rprouse) [inline src/NUnitFramework/tests/Internal/PlatformDetectionTests.cs]: This is a rather lame test for the fix, but it allowed me to put a break point in the code. I think the fixes for #869 are actually better indicators of the fix for #515. All tests are now passing on MacOSX, this one is getting hit and I am waiting on CI to test Linux and Windows.

**Comment by rprouse** [comment]
Test failures are the typical, unrelated errors. I restarted Travis, but don't have permission to do so with AppVeyor, it is under your account @CharliePoole.

File: src/NUnitFramework/tests/Internal/PlatformDetectionTests.cs (hunk 1 above)
Comment (CharliePoole) [inline src/NUnitFramework/tests/Internal/PlatformDetectionTests.cs]: Stylistically, I'd remove the Is.True. :-)

**Comment by CharliePoole** [comment]
Did you test this on CF?

**Comment by CharliePoole** [comment]
I restarted the Appveyor build and also added you as an Administrator.

**Comment by CharliePoole** [comment]
Looks ready to merge AFAIC.

**Comment by oznetmaster** [comment]
This does not build under CF.

OSPlatform.CurrentPlatform.IsMacOSX is not defined for CF build, but is called in the modified tests.

**Pull request merged**